syntax = "proto3";

package protocol;

import "databatch.proto";
import "eof_message.proto";
import "table_data.proto";
import "query_result.proto";
import "clean_up_message.proto";

enum MessageType {
  UNKNOWN = 0;
  DATA_BATCH = 1;
  EOF_MESSAGE = 2;
  TABLE_DATA = 3;
  QUERY_RESULT = 4;
  CLEAN_UP_MESSAGE = 5;
}


message Envelope {
  MessageType type = 1;

  oneof payload {
    DataBatch data_batch = 2;
    EOFMessage eof = 3;
    TableData table = 4;
    QueryResult query_result = 5;
    CleanUpMessage clean_up = 6;
  }
}
